Build and test errors if TILEDB_TBB=OFF is set

```
/home/jacobbolewski/TileDB/tiledb/sm/storage_manager/storage_manager.cc: In member function ‘tiledb::sm::Status tiledb::sm::StorageManager::init_tbb(tiledb::sm::Config::SMParams&)’:
/home/jacobbolewski/TileDB/tiledb/sm/storage_manager/storage_manager.cc:1674:51: warning: unused parameter ‘config’ [-Wunused-parameter]
 Status StorageManager::init_tbb(Config::SMParams& config) {
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,12 @@ endif()
 # Set C++11 as required standard for all C++ targets.
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF) # Don't use GNU extensions
+if(CMAKE_SYSTEM_NAME MATCHES "CYGWIN")
+  # Use GNU extensions under Cygwin
+  set(CMAKE_CXX_EXTENSIONS ON)
+else()
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
 
 # Set -fvisibility=hidden (or equivalent) flags by default.
 set(CMAKE_C_VISIBILITY_PRESET hidden)

--- a/tiledb/sm/misc/constants.cc
+++ b/tiledb/sm/misc/constants.cc
@@ -225,7 +225,7 @@ const uint64_t num_async_threads = 1;
 #ifdef HAVE_TBB
 const int num_tbb_threads = tbb::task_scheduler_init::automatic;
 #else
-const int num_tbb_threads = std::thread::hardware_concurrency();
+const int num_tbb_threads = -1;
 #endif
 
 /** The tile cache size. */

--- a/tiledb/sm/storage_manager/storage_manager.cc
+++ b/tiledb/sm/storage_manager/storage_manager.cc
@@ -716,8 +716,9 @@ Status StorageManager::init(Config* config) {
   tile_cache_ = new LRUCache(sm_params.tile_cache_size_);
   vfs_ = new VFS();
   RETURN_NOT_OK(vfs_->init(config_.vfs_params()));
+#ifdef HAVE_TBB
   RETURN_NOT_OK(init_tbb(sm_params));
-
+#endif
   auto& global_state = global_state::GlobalState::GetGlobalState();
   RETURN_NOT_OK(global_state.initialize(config));
   global_state.register_storage_manager(this);
@@ -1671,13 +1672,13 @@ Status StorageManager::get_fragment_uris(
   return Status::Ok();
 }
 
-Status StorageManager::init_tbb(Config::SMParams& config) {
 #ifdef HAVE_TBB
+Status StorageManager::init_tbb(Config::SMParams& config) {
   tbb_scheduler_ = std::unique_ptr<tbb::task_scheduler_init>(
       new tbb::task_scheduler_init(config.num_tbb_threads_));
-#endif
   return Status::Ok();
 }
+#endif
 
 Status StorageManager::load_array_schema(
     const URI& array_uri, ObjectType object_type, OpenArray* open_array) {

--- a/tiledb/sm/storage_manager/storage_manager.h
+++ b/tiledb/sm/storage_manager/storage_manager.h
@@ -798,6 +798,7 @@ class StorageManager {
   /** Increment the count of in-progress queries. */
   void increment_in_progress();
 
+#ifdef HAVE_TBB
   /**
    * Configures the TBB runtime. If TBB is not enabled, does nothing.
    *
@@ -805,6 +806,7 @@ class StorageManager {
    * @return Status
    */
   Status init_tbb(Config::SMParams& config);
+#endif
 
   /**
    * Loads the array schema into an open array.

